via-router/src/cache.rs

```diff
@@ -34,37 +29,33 @@ impl Cache {
         }
     }
 
-    pub fn read(&self, path: &str) -> Option<Option<Vec<Option<Match>>>> {
-        let lock = &self.entries;
-        let cap = self.capacity;
-
-        #[allow(clippy::never_loop)]
-        let (key, matches) = loop {
-            return match lock.try_read() {
-                Ok(guard) => match fetch(&guard, path) {
-                    Some(existing) => break existing,
-                    None => Some(None),
-                },
-                Err(_) => None,
-            };
-        };
-
-        if cap.checked_div(2).is_some_and(|half| key > half) {
```

Comment:
This business logic is now the responsibility of the caller with the introduction of `Cache::promote`.